src/test/java/frc/robot/util/SWMTest.java

```diff
@@ -0,0 +1,113 @@
+package frc.robot.util;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import frc.robot.commands.Shoot;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.subsystems.gpm.Shooter;
+import frc.robot.subsystems.gpm.StorageIndex;
+import frc.robot.subsystems.gpm.Arm;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SWMTest {
+    public static Shoot sh = new Shoot(new Shooter(), new Arm(), new Drivetrain(null), new StorageIndex());
+    public static int idx = 0;
+    /*
+     * make drivetrain in the test
+     * method to set pose, reset pose
+     * set chassis speed or drive() to make x,y velocities.
+     */
+    public double[][] test_cases = {
```

Comment:
where did you get these values

Reply:
I put them in myself. They show an initial state. Whether they work is at the mercy of the go() function's kinematics, and they do work.